KeyValue.cpp

```diff
@@ -658,10 +662,23 @@ void KeyValue::BuildData(char*& destBuffer)
 		}
 		else
 		{
-			memcpy(destBuffer, current->data.leaf.value.string, current->data.leaf.value.length);
-			destBuffer[current->data.leaf.value.length] = '\0';
+			char* string = current->data.leaf.value.string;
+			size_t size = current->data.leaf.value.length;
+			for ( int j = 0; j < size; j++ )
+			{
+				char c = *string++;
+				if ( c == '\\' && ( *string == '\\' || *string == '"' ) )
```

Comment:
Nitpick: this is a great opportunity to make a temporary value to simplify things and cut down on a dereference. Maybe something like a char named `peek` or `next` to represent the next character. The more we dereference, the worse our performance, and the worse our code is to read and understand.